src/OpenColorIO/Config.cpp

```diff
@@ -1078,6 +1094,435 @@ class Config::Impl
+            }
+
+            if (!EqualWithAbsError(vals[i], out[i], 1e-3f))
+            {
+                return "";
+            }
+        } 
+
+        //
+        // Then try the various primaries for the reference space.
+        //
+
+        // Define a (somewhat arbitrary) set of RGB values to test whether the transform is in fact 
+        // converting sRGB texture values to the candidate reference space. It includes 0.02 which is 
+        // on the sRGB linear segment, color values, and neutral values.
+        vals = { 0.7f,  0.4f,  0.02f,
+                 0.02f, 0.6f,  0.2f,
+                 0.3f,  0.02f, 0.5f,
+                 0.f,   0.f,   0.f,
+                 1.f,   1.f,   1.f };
+        std::string refSpace = "";
+        ConstProcessorRcPtr fromRefProc;
+        if (toRefTransform)
+        {
+            // The color space has the sRGB non-linearity. Now try combining the transform with a 
+            // transform from the Built-in config that goes from a variety of reference spaces to an 
+            // sRGB texture space. If the result is an identity, then that tells what the source config
+            // reference space is.
+            for (size_t i = 0; i < builtinLinearSpaces.size(); i++)
+            {
+                fromRefProc = getRefToSRGBTransform(builtinConfig, builtinLinearSpaces[i]);
+
+                ConstProcessorRcPtr toRefProc = getProcessorWithoutCaching(*config,
+                                                                           toRefTransform,
+                                                                           TRANSFORM_DIR_FORWARD);
+
+                ProcessorRcPtr proc = Processor::Create();
+                proc->getImpl()->concatenate(toRefProc, fromRefProc);
+
+                if (isIdentityTransform(proc, vals))
+                {
+                    refSpace = builtinLinearSpaces[i];
+                }
+            }
+        }
+
+        return refSpace;
+    }
+
+    ConstProcessorRcPtr getProcessorToBuiltinCS(ConstConfigRcPtr srcConfig,
+                                                const char * srcColorSpaceName, 
+                                                const char * builtinColorSpaceName,
+                                                TransformDirection direction) const
+    {
+        // Use the Default config as the Built-in config to interpret the known color space name.        
+        ConstConfigRcPtr builtinConfig = Config::CreateFromFile("ocio://default");
+
+        // Define the set of candidate reference linear color spaces (aka, reference primaries) that 
+        // will be used when searching through the source config. If the source config scene-referred 
+        // reference space is the equivalent of one of these spaces, it should be possible to identify 
+        // it with the following heuristics.
+        std::vector<std::string> builtinLinearSpaces = { "ACES - ACES2065-1", 
```

Comment:
I was debating about that.  I was hesitant to include even P3 since it seems an unlikely choice.  Are we aware of anyone using Rec.2020?  If so, we should certainly add it.

Reply:
Hum, I don't think I have an example on the top of my head. Maybe add it and make sure the order of search is from most to least common? Unless there are other drawbacks involved then I'm fine not including it.